pkg/component/component.go

```diff
@@ -645,15 +783,23 @@ func GetComponentSource(client *occlient.Client, componentName string, applicati
 }
 
 // Update updates the requested component
-// componentName is the name of the component to be updated
-// applicationName is the name of the application of the component
-// newSourceType indicates the type of the new source i.e git/local/binary
-// newSource indicates path of the source directory or binary or the git URL
-// stdout is the io writer for streaming build logs on stdout
-func Update(client *occlient.Client, componentName string, applicationName string, newSourceType string, newSource string, newSourceRef string, stdout io.Writer) error {
-
+// Parameters:
+//	client: occlient instance
+//	componentSettings: Component configuration
+//	newSource: Location of component source resolved to absolute path
+//	stdout: io pipe to write logs to
+// Returns:
+//	errors if any
+func Update(client *occlient.Client, componentSettings config.LocalConfigInfo, newSource string, stdout io.Writer) error {
 	// STEP 1. Create the common Object Meta for updating.
 
+	componentName := componentSettings.GetName()
+	applicationName := componentSettings.GetApplication()
+	newSourceType := componentSettings.GetSourceType()
+	newSourceRef := componentSettings.GetRef()
+	componentImageType := componentSettings.GetType()
+	cmpPorts := componentSettings.GetPorts()
```

Comment:
see previous comment in create.go, this is used again (retreiving all those values...)

if we're re-using code, this should be within a function returning a struct of information.

Reply:
https://github.com/openshift/odo/pull/1360#discussion_r268380836